e2e/pages/StageOne/ConsentPage.ts

```diff
@@ -0,0 +1,136 @@
+  readonly cancelButton: Locator;
+  readonly backButton: Locator;
+  // errors
+  readonly checkboxErrorMessage: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    // text
+    this.progressDisplay = page.getByText("31% complete");
+    this.consentPageHeading = page.getByRole("heading", {
+      name: "Consent to process your data and be contacted by Be Part of Research",
+    });
+    this.consentParagraph1 = page.getByText(
+      "By consenting and registering for the service, you are giving permission for any information you share to be processed by Be Part of Research, a service that is managed and operated by the National Institute for Health and Care Research Clinical Research Network Coordinating Centre (NIHR CRNCC). This will allow Be Part of Research to contact you with information about areas of research you have expressed an interest in, as well as research opportunities of national interest."
+    );
+    this.consentParagraph2 = page.getByText(
+      "To find out more about how the NIHR CRNCC will process and store your information and your rights as a data subject, please read the Be Part of Research Privacy Policy. The Data Controller for this service is the Department of Health and Social Care (DHSC)."
+    );
+    this.consentParagraph3 = page.getByText(
+      "You can withdraw your consent at any time - to do this please see the Be Part of Research Privacy Policy."
+    );
+    this.consentParagraph4 = page.getByText(
+      "If you do not consent to sharing your information, you will not be able to use this service to register your interest and be contacted about research opportunities."
+    );
+    this.consentParagraph5 = page.getByText(
+      "Do you give consent for your information to be processed by the Be Part of Research service provided by NIHR CRNCC on behalf of the DHSC for the purposes outlined above?"
+    );
+    // links
+    this.bePartOfResearchLink1 = page
+      .locator('a:text("Be Part of Research Privacy Policy")')
+      .nth(0);
+    this.bePartOfResearchLink2 = page
+      .locator('a:text("Be Part of Research Privacy Policy")')
+      .nth(1);
+    this.bePartOfResearchLink3 = page
+      .locator('a:text("Be Part of Research Privacy Policy")')
+      .nth(2);
+
+    // form
+    this.confirmCheckbox = page.locator("input#checkbox-1");
+    this.confirmMessage = page.getByText(
+      "I confirm that I have read and understood the Be Part of Research Privacy Policy."
+    );
+    //buttons
+    this.registerButton = page.getByRole("button", {
+      name: "Yes, I consent and wish to register now",
+    });
+    this.cancelButton = page.getByText(
+      "No, I do not consent and wish to cancel this registration"
+    );
+    this.backButton = page.getByTitle("Return to previous page");
+    // errors
+    this.checkboxErrorMessage = page.locator("span#checkbox--error-message");
+  }
+
+  // LOAD METHODS
+  async waitForPageLoad() {
+    await this.page.waitForSelector(
```

Comment:
Don't use the page.waitForSelector method - https://playwright.dev/docs/api/class-page#page-wait-for-selector
It's been replaced with the much more reliable locator.waitFor method - https://playwright.dev/docs/api/class-locator#locator-wait-for
You have already declared the same locator above, this.consentPageHeading, so use that

Or you could use your assertComponentsVisible method and pass it in
Or even just expect( this.consentPageHeading).toBeVisible()